src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -188,7 +188,7 @@ private void CheckInvariants()
             var branchId = GetBranchId();
 
             solutionInfo = solutionInfo ?? _solutionInfo;
-            projectIds = projectIds ?? _projectIds;
+            if (projectIds == default(ImmutableArray<ProjectId>)) { projectIds = _projectIds; }
```

Comment:
❕ Don't leave blocks on one line. Use one of the following instead:

Ternary expression:

```csharp
projectIds = !projectIds.IsDefault ? projectIds : _projectIds;
```

Expanded block:

```csharp
if (projectIds.IsDefault)
{
    projectIds = _projectIds;
}
```